src/librbd/internal.cc

```diff
@@ -2872,6 +2872,55 @@ int validate_mirroring_enabled(ImageCtx *ictx) {
             << cpp_strerror(r) << dendl;
         }
       }
+    } else if (next_mirror_mode == cls::rbd::MIRROR_MODE_DISABLED) {
+
+      bool needs_rollback = false;
+      if (current_mirror_mode == cls::rbd::MIRROR_MODE_POOL) {
+        r = cls_client::mirror_mode_set(&io_ctx, cls::rbd::MIRROR_MODE_IMAGE);
+        if (r < 0) {
+          lderr(cct) << "Failed to set mirror mode to image: "
+            << cpp_strerror(r) << dendl;
+          return r;
+        }
+        needs_rollback = true;
+      }
+
+      std::vector<std::string> image_ids;
+      r = cls_client::mirror_image_list(&io_ctx, &image_ids);
```

Comment:
Just to be clear, when you said:
> If you want all images to automatically be mirrored (assuming the images have journaling enabled), set it to image-mode
You wanted to say, "set it to pool-mode" right?

If that is the case, I agree with you, that we should prohibit "rbd mirror image enable/disable" when in pool-mode.